In [4]:
import pandas as pd
df = pd.read_csv("data.csv")


Lets see the details about the data set

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19735 entries, 0 to 19734
Data columns (total 29 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   date         19735 non-null  object 
 1   Appliances   19735 non-null  int64  
 2   lights       19735 non-null  int64  
 3   T1           19735 non-null  float64
 4   RH_1         19735 non-null  float64
 5   T2           19735 non-null  float64
 6   RH_2         19735 non-null  float64
 7   T3           19735 non-null  float64
 8   RH_3         19735 non-null  float64
 9   T4           19735 non-null  float64
 10  RH_4         19735 non-null  float64
 11  T5           19735 non-null  float64
 12  RH_5         19735 non-null  float64
 13  T6           19735 non-null  float64
 14  RH_6         19735 non-null  float64
 15  T7           19735 non-null  float64
 16  RH_7         19735 non-null  float64
 17  T8           19735 non-null  float64
 18  RH_8         19735 non-null  float64
 19  T9  

Data contains 29 columns for variables, the first column is a datetime, which means its a time series data


The variable names are a bit confusing so i;ll try to write these names clearly


__Datetime__    &emsp; : Date and time in Year-Month-Day Hour:Minute:Second

__EnergyApp_Wh__ &emsp;: Energy use in Watthours

__lights_Wh__ &emsp;: energy use of light fixtures in the house in Wh


__T1_kitchen__ &emsp; : Temperature in kitchen area


__Hum1_kitchen__ : Humidity in Kitchen

__T2_living__ : Temperature living room


__Hum2_living__ : Humidity living room
 
 
__T3_laundry__ : Temperature in laundry room area

__Hum3_Laundry__ : Humidity in laundry room area

__T4_office__ :  Temperature in office room in Celsius

__Hum4_office__ :  Humidity in office room in %


__T5_bathroom__: Temperature in bathroom, in Celsius

__Hum5_bathroom__: Humidity in bathroom, in %


__T6_outside_north__:

__Hum6_outside__:

__T7_iron_room__:

__Hum7_iron__:

__T8_teen_room__:

__Hum8_teen_room__:

__T9_parents__:

__Hum9_parents__:

__T10_outside_WS__:

__Pressure_mmHg__:


__Hum_outside_WS__:

__WindSpeed_WS_m/s__:

__Visibility_kilometer__:

__T_dew_point__:

__random_1__:

__random_2__:



In [6]:
header = ["datetime", "EnergyApp_Wh", "lights_Wh", "T1_kitchen", "Hum1_kitchen", "T2_living", "Hum2_living", "T3_laundry", "Hum3_Laundry", "T4_office", "Hum4_office", "T5_bathroom", "Hum5_bathroom", "T6_outside_north", "Hum6_outside", "T7_iron_room", "Hum7_iron", "T8_teen_room", "Hum8_teen_room", "T9_parents", "Hum9_parents", "T10_outside_WS", "Pressure_mmHg", "Hum_outside_WS", "WindSpeed_WS_m/s", "Visibility_kilometer", "T_dew_point", "random_1", "random_2"]
header

['datetime',
 'EnergyApp_Wh',
 'lights_Wh',
 'T1_kitchen',
 'Hum1_kitchen',
 'T2_living',
 'Hum2_living',
 'T3_laundry',
 'Hum3_Laundry',
 'T4_office',
 'Hum4_office',
 'T5_bathroom',
 'Hum5_bathroom',
 'T6_outside_north',
 'Hum6_outside',
 'T7_iron_room',
 'Hum7_iron',
 'T8_teen_room',
 'Hum8_teen_room',
 'T9_parents',
 'Hum9_parents',
 'T10_outside_WS',
 'Pressure_mmHg',
 'Hum_outside_WS',
 'WindSpeed_WS_m/s',
 'Visibility_kilometer',
 'T_dew_point',
 'random_1',
 'random_2']

In [7]:
df = pd.read_csv("data.csv")


df.columns = header
df.head()

datetime  EnergyApp_Wh  lights_Wh  T1_kitchen  Hum1_kitchen  \
0  2016-01-11 17:00:00            60         30       19.89     47.596667   
1  2016-01-11 17:10:00            60         30       19.89     46.693333   
2  2016-01-11 17:20:00            50         30       19.89     46.300000   
3  2016-01-11 17:30:00            50         40       19.89     46.066667   
4  2016-01-11 17:40:00            60         40       19.89     46.333333   

   T2_living  Hum2_living  T3_laundry  Hum3_Laundry  T4_office  ...  \
0       19.2    44.790000       19.79     44.730000  19.000000  ...   
1       19.2    44.722500       19.79     44.790000  19.000000  ...   
2       19.2    44.626667       19.79     44.933333  18.926667  ...   
3       19.2    44.590000       19.79     45.000000  18.890000  ...   
4       19.2    44.530000       19.79     45.000000  18.890000  ...   

   T9_parents  Hum9_parents  T10_outside_WS  Pressure_mmHg  Hum_outside_WS  \
0   17.033333         45.53        6.600000          733.5            92.0   
1   17.066667         45.56        6.483333          733.6            92.0   
2   17.000000         45.50        6.366667          733.7            92.0   
3   17.000000         45.40        6.250000          733.8            92.0   
4   17.000000         45.40        6.133333          733.9            92.0   

   WindSpeed_WS_m/s  Visibility_kilometer  T_dew_point   random_1   random_2  
0          7.000000             63.000000          5.3  13.275433  13.275433  
1          6.666667             59.166667          5.2  18.606195  18.606195  
2          6.333333             55.333333          5.1  28.642668  28.642668  
3          6.000000             51.500000          5.0  45.410389  45.410389  
4          5.666667             47.666667          4.9  10.084097  10.084097  

[5 rows x 29 columns]

In [8]:
df.info()

df.describe()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19735 entries, 0 to 19734
Data columns (total 29 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   datetime              19735 non-null  object 
 1   EnergyApp_Wh          19735 non-null  int64  
 2   lights_Wh             19735 non-null  int64  
 3   T1_kitchen            19735 non-null  float64
 4   Hum1_kitchen          19735 non-null  float64
 5   T2_living             19735 non-null  float64
 6   Hum2_living           19735 non-null  float64
 7   T3_laundry            19735 non-null  float64
 8   Hum3_Laundry          19735 non-null  float64
 9   T4_office             19735 non-null  float64
 10  Hum4_office           19735 non-null  float64
 11  T5_bathroom           19735 non-null  float64
 12  Hum5_bathroom         19735 non-null  float64
 13  T6_outside_north      19735 non-null  float64
 14  Hum6_outside          19735 non-null  float64
 15  T7_iron_room       

EnergyApp_Wh     lights_Wh    T1_kitchen  Hum1_kitchen     T2_living  \
count  19735.000000  19735.000000  19735.000000  19735.000000  19735.000000   
mean      97.694958      3.801875     21.686571     40.259739     20.341219   
std      102.524891      7.935988      1.606066      3.979299      2.192974   
min       10.000000      0.000000     16.790000     27.023333     16.100000   
25%       50.000000      0.000000     20.760000     37.333333     18.790000   
50%       60.000000      0.000000     21.600000     39.656667     20.000000   
75%      100.000000      0.000000     22.600000     43.066667     21.500000   
max     1080.000000     70.000000     26.260000     63.360000     29.856667   

        Hum2_living    T3_laundry  Hum3_Laundry     T4_office   Hum4_office  \
count  19735.000000  19735.000000  19735.000000  19735.000000  19735.000000   
mean      40.420420     22.267611     39.242500     20.855335     39.026904   
std        4.069813      2.006111      3.254576      2.042884      4.341321   
min       20.463333     17.200000     28.766667     15.100000     27.660000   
25%       37.900000     20.790000     36.900000     19.530000     35.530000   
50%       40.500000     22.100000     38.530000     20.666667     38.400000   
75%       43.260000     23.290000     41.760000     22.100000     42.156667   
max       56.026667     29.236000     50.163333     26.200000     51.090000   

       ...    T9_parents  Hum9_parents  T10_outside_WS  Pressure_mmHg  \
count  ...  19735.000000  19735.000000    19735.000000   19735.000000   
mean   ...     19.485828     41.552401        7.411665     755.522602   
std    ...      2.014712      4.151497        5.317409       7.399441   
min    ...     14.890000     29.166667       -5.000000     729.300000   
25%    ...     18.000000     38.500000        3.666667     750.933333   
50%    ...     19.390000     40.900000        6.916667     756.100000   
75%    ...     20.600000     44.338095       10.408333     760.933333   
max    ...     24.500000     53.326667       26.100000     772.300000   

       Hum_outside_WS  WindSpeed_WS_m/s  Visibility_kilometer   T_dew_point  \
count    19735.000000      19735.000000          19735.000000  19735.000000   
mean        79.750418          4.039752             38.330834      3.760707   
std         14.901088          2.451221             11.794719      4.194648   
min         24.000000          0.000000              1.000000     -6.600000   
25%         70.333333          2.000000             29.000000      0.900000   
50%         83.666667          3.666667             40.000000      3.433333   
75%         91.666667          5.500000             40.000000      6.566667   
max        100.000000         14.000000             66.000000     15.500000   

           random_1      random_2  
count  19735.000000  19735.000000  
mean      24.988033     24.988033  
std       14.496634     14.496634  
min        0.005322      0.005322  
25%       12.497889     12.497889  
50%       24.897653     24.897653  
75%       37.583769     37.583769  
max       49.996530     49.996530  

[8 rows x 28 columns]

In [9]:
# Dataframe shape is
df.shape
rows = df.shape[0]
col = df.shape[1]

In [10]:
# Check for null values
df.isnull().sum().sort_values(ascending=True)

# No null values good to go!

datetime                0
T_dew_point             0
Visibility_kilometer    0
WindSpeed_WS_m/s        0
Hum_outside_WS          0
Pressure_mmHg           0
T10_outside_WS          0
Hum9_parents            0
T9_parents              0
Hum8_teen_room          0
T8_teen_room            0
Hum7_iron               0
T7_iron_room            0
random_1                0
Hum6_outside            0
Hum5_bathroom           0
T5_bathroom             0
Hum4_office             0
T4_office               0
Hum3_Laundry            0
T3_laundry              0
Hum2_living             0
T2_living               0
Hum1_kitchen            0
T1_kitchen              0
lights_Wh               0
EnergyApp_Wh            0
T6_outside_north        0
random_2                0
dtype: int64

In [11]:
# As shown above there are no null values

In [12]:
# we are going to ignore the date column

# let's merge together same types of features

house_temp = ["T1_kitchen", "T2_living", "T3_laundry", "T4_office", "T5_bathroom", "T6_outside_north",
              "T7_iron_room", "T8_teen_room", "T9_parents"]

house_humidity = ["Hum1_kitchen", "Hum2_living", "Hum3_Laundry", "Hum4_office", "Hum5_bathroom",
                  "Hum6_outside", "Hum7_iron", "Hum8_teen_room", "Hum9_parents"]

weather_features = ["T10_outside_WS", "Pressure_mmHg", "Hum_outside_WS", "WindSpeed_WS_m/s",
                    "Visibility_kilometer", "T_dew_point"]

lights = ["lights_Wh"]


In [13]:
random_features = ["random_1", "random_2"]

target = ["EnergyApp_Wh"]

In [14]:

from sklearn.model_selection import train_test_split

train, test = train_test_split(df, test_size=0.3, random_state=42)

features = train[house_temp+house_humidity+weather_features+lights+random_features]
features

T1_kitchen  T2_living  T3_laundry  T4_office  T5_bathroom  \
9129    21.500000  17.790000   21.700000      20.39    19.600000   
2453    19.500000  18.700000   19.823333      18.10    17.200000   
9152    20.790000  16.890000   21.700000      19.70    19.290000   
12694   22.100000  19.963333   23.390000      21.10    20.660000   
16952   24.700000  29.856667   26.171429      25.10    23.166667   
...           ...        ...         ...        ...          ...   
11284   21.760000  19.390000   23.000000      21.50    20.100000   
11964   22.390000  22.000000   22.700000      21.29    19.700000   
5390    20.290000  18.200000   20.100000      18.20    18.500000   
860     21.790000  20.434000   21.200000      21.10    18.050000   
15795   21.323333  19.890000   22.700000      19.70    19.200000   

       T6_outside_north  T7_iron_room  T8_teen_room  T9_parents  Hum1_kitchen  \
9129           0.133333         20.79     22.100000   19.463333     35.626667   
2453           3.390000         18.89     18.790000   17.200000     44.560000   
9152          -1.533333         20.39     21.730000   19.390000     35.400000   
12694          9.190000         21.33     24.390000   20.290000     43.260000   
16952         26.390000         24.00     24.434000   23.100000     42.360000   
...                 ...           ...           ...         ...           ...   
11284          9.500000         21.00     24.166667   20.356667     40.900000   
11964         17.893333         21.50     24.200000   20.290000     43.700000   
5390          -2.090000         18.50     21.260000   17.890000     35.700000   
860            1.200000         21.79     22.700000   18.500000     35.560000   
15795         10.890000         19.10     21.290000   19.390000     37.730000   

       ...  Hum9_parents  T10_outside_WS  Pressure_mmHg  Hum_outside_WS  \
9129   ...     38.260000        0.250000     766.400000       83.000000   
2453   ...     46.163333        3.166667     765.266667       85.333333   
9152   ...     39.067500       -1.566667     766.000000       89.333333   
12694  ...     37.400000        8.833333     753.366667       81.000000   
16952  ...     44.466667       21.433333     752.100000       51.000000   
...    ...           ...             ...            ...             ...   
11284  ...     40.560000        9.566667     750.466667       77.666667   
11964  ...     44.290000       17.300000     749.500000       66.000000   
5390   ...     39.466667       -1.066667     754.900000       87.000000   
860    ...     41.466667        2.066667     765.033333       64.333333   
15795  ...     41.266667        6.650000     758.650000       85.500000   

       WindSpeed_WS_m/s  Visibility_kilometer  T_dew_point  lights_Wh  \
9129           2.000000             65.000000    -2.350000          0   
2453           2.000000             40.000000     0.966667          0   
9152           1.333333             60.666667    -3.100000          0   
12694          1.666667             26.000000     5.733333          0   
16952          2.000000             40.000000    10.800000          0   
...                 ...                   ...          ...        ...   
11284          6.666667             40.000000     5.833333          0   
11964          2.000000             40.000000    10.900000          0   
5390           3.666667             62.666667    -2.933333          0   
860            1.333333             40.000000    -4.066667          0   
15795          1.000000             40.000000     4.350000          0   

        random_1   random_2  
9129   36.226675  36.226675  
2453   43.199767  43.199767  
9152   24.976055  24.976055  
12694  16.161125  16.161125  
16952  17.055346  17.055346  
...          ...        ...  
11284  45.513197  45.513197  
11964  34.934752  34.934752  
5390   19.071358  19.071358  
860     1.050592   1.050592  
15795  17.456040  17.456040  

[13814 rows x 27 columns]

In [ ]:
target_feature = train[target]

feat_description = features.describe()
feat_description

a few observations here
The minimum temp inside the house is 14.89 degree celsius
The maximum temp inside the house is 28.29 degree celsius

humidity varies between 20.60% to 63.36% with exception of bathroom

In [18]:
target_description = target_feature.describe()

# according to target description 75 % of the appliances consume 100 watt or less

features['lights_Wh'].value_counts()
# there are 10685 zeros in total 13814 observations, this means that
# lights are off most of the time, so we are going to ignore them

0     10685
10     1543
20     1121
30      399
40       57
50        7
70        1
60        1
Name: lights_Wh, dtype: int64

In [ ]:

features.drop('lights_Wh', axis=1, inplace=True)

c:\users\fahad ali sarwar\pycharmprojects\energy\venv\lib\site-packages\pandas\core\frame.py:3997: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  errors=errors,
